client/asset/dcr/dcr.go

```diff
@@ -2846,6 +2942,60 @@ func (dcr *ExchangeWallet) getBestBlock(ctx context.Context) (*block, error) {
 	return &block{hash: hash, height: height}, nil
 }
 
+func (dcr *ExchangeWallet) getBlock(ctx context.Context, blockHash *chainhash.Hash, verboseTx bool) (*chainjson.GetBlockVerboseResult, error) {
+	blockVerbose, err := dcr.wallet.GetBlockVerbose(ctx, blockHash, verboseTx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving block %s: %w", blockHash, translateRPCCancelErr(err))
+	}
+	return blockVerbose, nil
+}
+
+func (dcr *ExchangeWallet) getBlockHash(blockHeight int64) (*chainhash.Hash, error) {
+	blockHash, err := dcr.wallet.GetBlockHash(dcr.ctx, blockHeight)
+	if err != nil {
+		return nil, translateRPCCancelErr(err)
```

Comment:
The `dcr.Wallet` interface implementations can be responsible for translating into an `asset.ErrRequestTimeout`.  Looks like the `rpcWallet` methods are doing it already.

As a bonus, this will reduce some of these to oneliners.